src/openssl/algorithm.cr

```diff
@@ -0,0 +1,35 @@
+require "openssl"
+
+module OpenSSL
+  enum Algorithm
+    # This enum indicates the possible hash algorithms for *OpenSSL* digest operations.
+    MD4
+    MD5
+    RIPEMD160
+    SHA1
+    SHA224
+    SHA256
+    SHA384
+    SHA512
+
+    def to_evp
+      # The internal bindings to the *LibCrypto* digest operations sometimes require a hash algorithm
```

Comment:
Ditto. Needs to be put in front of `to_evp`.

Reply:
"equivalent hash algorithm" should be more precise. This enum already represents hash algorithms. This method returns the corresponding identifier in libcrypto.

Reply:
The @r00ster91's suggestion seems very concise to me. Thanks for the suggestion.
But really have to agree that this phrasing adds some ambiguity like stated by @straight-shoota.

Reply:
The *OpenSSL* documentation mentions the expression "message digest function" when referring to this type.

Also, the documentation for the *SSLeay* library (from where *OpenSSL* was forked) makes it clear that the acronym **EVP** stands for Envelope.

Reply:
The type of the _equivalent hash algorithms_ in the _LibCrypto API_ is `EVP_MD`.

I found that this `MD` stands for _Message Digest_, and in some point it mentions the expressions and _"digest method"_ and _"message digest algorithm"_ related to this type.

That pretty much confuses me. I thought that the algorithm was called **hash** and the ouput was called **digest**. ([source](https://crypto.stackexchange.com/questions/51243/what-is-the-difference-between-a-digest-and-a-hash-function)).

It seems that back there _digest algorithm_ meant what now we call _hash algorithm_.

Well... I think this is no help at all. I just wanted to add some value, as I can't by means of language, since I do not write very well in english.

Reply:
What about this review? 
I'm not sure what to do